std/str.zig

```diff
@@ -10,6 +11,15 @@ pub fn slice_eql(T: type)(a: []const T, b: []const T) -> bool {
     return true;
 }
 
+pub fn join(T: type)(a: []T, b: []T) -> []T {
+    var result : [a.len + b.len]T = undefined;
+
+    @memcpy(result.ptr, a.ptr, a.len * @sizeof(T));
+    @memcpy((&T)((usize)(result.ptr) + (usize)(a.len * @sizeof(T))), b.ptr, b.len * @sizeof(T));
```

Comment:
I think use of `@compile_var` like this is a code smell.

I think the solution to this is to provide a higher level memcpy that operates on slices. For example:

```zig
fn copy(T: type)(dest: []T, src: []T) {
    assert(dest.len >= src.len);
    // ...
}
```

Then you could do:

```zig
copy(T)(result, a);
copy(T)(result[a.len...], b);
```